### PR TITLE
add boot order to network interface for 0.6

### PR DESCRIPTION
**Pull request by phoracek**
(cherry picked from commit af75d4882003eea472b5c6fc9501b4f797e933e2)

**What this PR does / why we need it**:
Backport of #1411. With this patch and L2 network plugin, we make it possible to boot VM from PXE.

**Special notes for your reviewer**:
Not sure whether we want to merge this. Is it possible to use it with default Kubernetes network?

### CODE REVIEW

**Comment by rmohr** [comment]
ok to test

**Comment by rmohr** [comment]
retest this please

File: pkg/virt-api/validating-webhook/validating-webhook.go [hunk 1]
```diff
@@ -500,6 +516,28 @@ func ValidateVirtualMachineInstanceSpec(field *k8sfield.Path, spec *v1.VirtualMa
 					Field:   field.Child("domain", "devices", "networks").Index(0).Child("pod").String(),
 				})
 			}
+
+			if iface.BootOrder != nil {
+				order := *iface.BootOrder
+				// Verify boot order is greater than 0, if provided
+				if order < 1 {
```
Comment (stu-gott) [inline pkg/virt-api/validating-webhook/validating-webhook.go]: Perhaps a dumb question, but isn't boot order defined as an unsigned int? Why not just allow `0` as a valid order index and not worry about it?

**Comment by stu-gott** [comment]
ci test please

File: pkg/virt-api/validating-webhook/validating-webhook.go (hunk 1 above)
Comment (yuvalif) [reply pkg/virt-api/validating-webhook/validating-webhook.go]: order "0" is not valid in libvirt.

File: pkg/virt-launcher/virtwrap/api/converter.go [hunk 2]
```diff
@@ -538,6 +538,11 @@ func Convert_v1_VirtualMachine_To_api_Domain(vmi *v1.VirtualMachineInstance, dom
 		return nil, fmt.Errorf("failed to find network %s", name)
 	}
 
+	networks := map[string]*v1.Network{}
```
Comment (yuvalif) [inline pkg/virt-launcher/virtwrap/api/converter.go]: this is unrelated change. is this a bug fix in 0.6?

**Comment by cynepco3hahue** [comment]
ci test please

**Comment by cynepco3hahue** [comment]
ci test please

**Comment by cynepco3hahue** [comment]
Tests look pretty good, OS jobs have 4 failures related to storage, but I pretty sure it does not because of this PR.
